bluebutton-data-pipeline-fhir-load/src/main/java/gov/hhs/cms/bluebutton/datapipeline/fhir/transform/DataTransformer.java

```diff
@@ -1559,6 +1577,172 @@ private TransformedBundle transformInpatientClaim(RifRecordEvent<InpatientClaimG
+		serviceProviderOrg.setType(new CodeableConcept().addCoding(new Coding()
+				.setSystem(CODING_SYSTEM_CCW_FACILITY_TYPE_CD).setCode(claimGroup.claimFacilityTypeCode.toString())));
+		Reference serviceProviderOrgReference = upsert(bundle, serviceProviderOrg,
+				referenceOrganizationByNpi(claimGroup.organizationNpi).getReference());
+		eob.setOrganization(serviceProviderOrgReference);
+
+		Practitioner attendingPhysician = new Practitioner();
+		attendingPhysician.addIdentifier().setSystem(CODING_SYSTEM_NPI_US).setValue(claimGroup.attendingPhysicianNpi);
+		Reference attendingPhysicianRef = referencePractitioner(claimGroup.attendingPhysicianNpi);
+		upsert(bundle, attendingPhysician, attendingPhysicianRef.getReference());
+
+		Practitioner operatingPhysician = new Practitioner();
+		operatingPhysician.addIdentifier().setSystem(CODING_SYSTEM_NPI_US).setValue(claimGroup.operatingPhysicianNpi);
+		Reference operatingPhysicianRef = referencePractitioner(claimGroup.operatingPhysicianNpi);
+		upsert(bundle, operatingPhysician, operatingPhysicianRef.getReference());
+
+		Practitioner otherPhysician = new Practitioner();
+		otherPhysician.addIdentifier().setSystem(CODING_SYSTEM_NPI_US)
+				.setValue(claimGroup.otherPhysicianNpi.toString());
+		Reference otherPhysicianRef = referencePractitioner(claimGroup.otherPhysicianNpi.toString());
+		upsert(bundle, otherPhysician, otherPhysicianRef.getReference());
+
+		/*
+		 * TODO once STU3 is available, transform financial/payment amounts into
+		 * eob.information entries
+		 */
+
+		addDiagnosisCode(eob, claimGroup.diagnosisPrincipal);
+		for (IcdCode diagnosis : claimGroup.diagnosesAdditional)
+			addDiagnosisCode(eob, diagnosis);
+
+		if (claimGroup.diagnosisFirstClaimExternal.isPresent()) {
+			addDiagnosisCode(eob, (claimGroup.diagnosisFirstClaimExternal.get()));
+		}
+
+		for (IcdCode diagnosis : claimGroup.diagnosesExternal)
+			if (!diagnosis.getCode().isEmpty()) {
+				addDiagnosisCode(eob, diagnosis);
+			}
+
+		for (IcdCode diagnosis : claimGroup.diagnosesReasonForVisit)
+			if (!diagnosis.getCode().isEmpty()) {
+				addDiagnosisCode(eob, diagnosis);
+			}
+
+		for (OutpatientClaimLine claimLine : claimGroup.lines) {
+			ItemsComponent item = eob.addItem();
+			item.setSequence(claimLine.lineNumber);
+
+			item.setType(new Coding().setSystem(CODING_SYSTEM_FHIR_EOB_ITEM_TYPE)
+					.setCode(CODED_EOB_ITEM_TYPE_CLINICAL_SERVICES_AND_PRODUCTS));
+
+			/*
+			 * TODO once STU3 available, transform
+			 * claimServiceClassificationTypeCode into eob.item.category.
+			 */
+			item.setService(new Coding().setSystem(CODING_SYSTEM_HCPCS).setCode(claimLine.hcpcsCode));
+
+			item.addAdjudication()
+					.setCategory(new Coding().setSystem(CODING_SYSTEM_ADJUDICATION_CMS)
+							.setCode(CODED_ADJUDICATION_BLOOD_DEDUCTIBLE))
+					.getAmount().setSystem(CODING_SYSTEM_MONEY).setCode(CODING_SYSTEM_MONEY_US)
+					.setValue(claimLine.bloodDeductibleAmount);

```

Comment:
Just a heads-up, that I expect a lot of these financial amounts will end up being nullable/optional, once Scott K gets us that analysis. You're correct to not worry about that yet-- just an FYI.